dsc/src/main.rs

```diff
@@ -54,16 +54,63 @@ fn main() {
             generate(shell, &mut cmd, "dsc", &mut io::stdout());
         },
         SubCommand::Config { subcommand, parameters, parameters_file, system_root, as_group, as_assert, as_include } => {
-            if let Some(file_name) = parameters_file {
+            let merged_parameters = if parameters_file.is_some() && parameters.is_some() {
+                // Both parameters and parameters_file provided - merge them with inline taking precedence
+                let file_params = if let Some(file_name) = &parameters_file {
+                    if file_name == "-" {
+                        info!("{}", t!("main.readingParametersFromStdin"));
+                        let mut stdin = Vec::<u8>::new();
+                        match io::stdin().read_to_end(&mut stdin) {
+                            Ok(_) => {
+                                match String::from_utf8(stdin) {
+                                    Ok(input) => Some(input),
+                                    Err(err) => {
+                                        error!("{}: {err}", t!("util.invalidUtf8"));
+                                        exit(EXIT_INVALID_INPUT);
+                                    }
+                                }
+                            },
+                            Err(err) => {
+                                error!("{}: {err}", t!("util.failedToReadStdin"));
+                                exit(EXIT_INVALID_INPUT);
+                            }
+                        }
```

Comment:
Reading from STDIN is duplicated here and lines below.  I think instead of this approach, you can always assume to merge and sometimes file will be empty or inline will be empty.